programs/fileio.c

```diff
@@ -1261,10 +1268,10 @@ FIO_compressZstdFrame(FIO_prefs_t* const prefs,
                                 (unsigned)(zfp.produced >> 20),
                                 cShare );
                 } else {   /* summarized notifications if == 2; */
-                    DISPLAYLEVEL(2, "\rRead : %u ", (unsigned)(zfp.consumed >> 20));
+                    DISPLAYLEVEL(2, "\033[s Read : %u ", (unsigned)(zfp.consumed >> 20));
```

Comment:
So far, we have avoided escape codes, due to portability concerns (unsure how this works on _all_ consoles).

I presume that if we want to avoid that risk,
existing design based on `\r` would require to rewrite the entire line, instead of just the count section.

Do you think this would be feasible ?

Reply:
Within `FIO_compressZstdFrame()` we don't have any idea which # file we're currently compressing, which we'd need to regenerate the `compressing file currentFile/totalFiles` part of the output line. We do have access to `FIO_prefs` which maybe we could use to store more information, like a `currFileIdx`. I'm inclined to go with this, since we already have added a `nbFiles` field.

If we're willing to forgo the `currentFile/totalFiles`, we can definitely just have it output the typical: `[sourcefilename] : [some info about the progress` and have it continually overwrite that. 

If we could separate it into two lines, that might be another way, but I haven't come across anything that lets me replace already-printed console lines that looks portable.